test/unit/url.test.js

```diff
@@ -1317,6 +1473,20 @@ describe('Url', function () {
                 expect(url.query.all()[1].key).to.eql('foo2  ');
                 expect(url.query.all()[1].value).to.eql('bar2  ');
             });
+
+            it('should handle query parameters with empty key or value', function () {
+                var url = new Url('https://postman-echo.com?get&=bar&=&baz=&&');
+
+                expect(url.query.all()).to.have.deep.members([
+                    new QueryParam({ key: 'get', value: null }),
+                    new QueryParam({ key: '', value: 'bar' }),
+                    new QueryParam({ key: '', value: '' }),
+                    new QueryParam({ key: 'baz', value: '' }),
+                    new QueryParam({ key: null, value: null }),
+                    // why's ☝🏻 different from 👇🏻 (╯°□°)╯︵ ┻━┻
```

Comment:
It is because of this: https://github.com/postmanlabs/postman-collection/pull/984/files#diff-e02ef5f2975cda047ffc0d6996253ae0R120

But why is it needed!? 🤔

Reply:
Yes, but don't want to change that flow right now.